components/user/UserEvents.js

```diff
@@ -1,46 +1,66 @@
 import { useState } from "react";
-
 import EventCard from "@components/event/EventCard";
 import Alert from "@components/Alert";
-import EventKey from "@components/event/EventKey";
 
 export default function UserEvents({ data }) {
-  const [eventType, seteventType] = useState("future");
-  const futureEvents = data.events.filter(
-    (event) => new Date(event.date.start) > new Date()
-  );
+  const [eventType, setEventType] = useState("all");
+  
+  const eventOptions = [
+    { value: 'all' , name: 'All Events'},
+    { value: 'future' , name: 'Future Events'},
+    { value: 'ongoing', name: 'Ongoing Events'},
+    { value: 'virtual', name: 'Virtual Events'},
+    { value: 'inPerson', name: 'In-Person Events'},
+    { value: 'cfpOpen', name:'Events with open CFP'},
+    { value: 'past', name: 'Past Events'}
+  ]
 
-  let categorisedEvents = {
-    future: futureEvents,
-    ongoing: data.events.filter(
-      (event) =>
-        new Date(event.date.start) < new Date() &&
-        new Date(event.date.end) > new Date()
-    ),
-    virtual: futureEvents.filter((event) => event.isVirtual === true),
-    inPerson: futureEvents.filter((event) => event.isInPerson === true),
-    cfpOpen: futureEvents.filter((event) =>
-      event.date.cfpClose ? new Date(event.date.cfpClose) > new Date() : false
-    ),
-    past: data.events
-      .filter((event) => new Date(event.date.end) < new Date())
-      .sort((a, b) => new Date(b.date.start) - new Date(a.date.start)),
+  const handleEventTypeChange = (event) => {
+    setEventType(event.target.value);
   };
 
+  const eventsToShow = eventType === "all"
+    ? data.events
+    : data.events.filter(event => {
+        const startDate = new Date(event.date.start);
+        const endDate = new Date(event.date.end);
+        const now = new Date();
+        return eventType === "future" && startDate > now
+          || eventType === "ongoing" && startDate <= now && now <= endDate
+          || eventType === "virtual" && startDate > now && event.isVirtual
+          || eventType === "inPerson" && startDate > now && event.isInPerson
+          || eventType === "cfpOpen" && startDate > now && event.date.cfpClose > now
+          || eventType === "past" && endDate < now;
```

Comment:
Having a single filter makes it very difficult to read, I suggest splitting these up

Reply:
@eddiejaoude "Thank you for your valuable suggestions! I have made the changes you suggested:

1. Added more spacing when there are no results.
2. Aligned the label and select horizontally instead of stacking them vertically.
3. Split the single filter for better readability.

However, I am confused about 
> We have `form/Input.js`, I think the `select` could be moved there also to its own component

 I would like to clarify your suggestion. Are you suggesting that we create a separate component for select and name it select.js within the form folder?

Reply:
> I would like to clarify your suggestion. Are you suggesting that we create a separate component for select and name it select.js within the form folder?

Yes I think this would be a good idea 👍 sorry if I did not explain it well

Something similar to the way `Input.js` is done

Reply:
I have refactored the code to replace the **select** element with the **DropDown** component that I created in `form` folder. This will make the code more readable, concise, and reusable I guess. Any further suggestions @eddiejaoude 

 Before creating ** `DropDown.js`**  in form folder **

 ````
<label
        htmlFor="event-type-select"
        className=" mr-2 font-semibold text-gray-700"
      >
        Select Event Type:
      </label>
      <select
        id="event-type-select"
        value={eventType}
        onChange={handleEventTypeChange}
        className="block w-full max-w-lg px-4 py-2 border border-orange-300 rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-orange-500 focus:border-orange-500"
      >
        {eventOptions.map(option => (
          <option key={option.value} value={option.value}>{option.name}</option>
        ))}
</select>
````

After creating `DropDown.js` Component

````
<DropdownMenu
      eventType={eventType}
      handleEventTypeChange={handleEventTypeChange} 
      options={eventOptions} 
      label="Select Event Type:"
      className="inline text-center text-sm font-medium leading-6 text-gray-900 sm:pt-1.5" 
      />